### PR TITLE
Add `--probe-args` for TCP SYN module to send packets with the various sets of TCP options

### PR DESCRIPTION
**Pull request by phillip-stephens**
It seems that various hitrates are achieved by changing the TCP options each packet is sent with. This PR adds the following options:

- Same options as linux (Ubuntu 23.04)
- BSD (macOS Sonoma)
- Windows (Windows 11)
- No Options

The `--help` text for the module is also updated to reflect the new options.
Options from different OS's, collected with Wireshark
image
`none` Options
image
Ubuntu(`linux`) Options
image
`bsd` (macOS) Options
image
`Windows` (Windows 11)
image

### CODE REVIEW

File: run_tests.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
+import os
```
Comment (phillip-stephens) [inline run_tests.py:1]: Remove this file

File: src/probe_modules/module_tcp_synscan.c [hunk 2]
```diff
@@ -16,22 +16,60 @@
...
 
 static int synscan_global_initialize(struct state_conf *state)
 {
 	num_source_ports =
 	    state->source_port_last - state->source_port_first + 1;
+	module_tcp_synscan.max_packet_length = ZMAP_TCP_SYNSCAN_PACKET_LEN;
+	    // Based on the OS, we'll set the TCP options differently
+	if (!state->probe_args) {
+		// user didn't provide any probe args, defaulting to linux
+		log_debug("tcp_synscan", "no probe-args, defaulting to linux-like TCP options");
```
Comment (phillip-stephens) [inline src/probe_modules/module_tcp_synscan.c:43]:
```suggestion
		log_debug("tcp_synscan", "no probe-args, defaulting to linux TCP options");
```

File: src/probe_modules/module_tcp_synscan.c [hunk 3]
```diff
@@ -225,10 +262,14 @@ probe_module_t module_tcp_synscan = {
...
-    .helptext = "Probe module that sends a TCP SYN packet to a specific "
-		"port. Possible classifications are: synack and rst. A "
-		"SYN-ACK packet is considered a success and a reset packet "
-		"is considered a failed response.",
+    .helptext =
+	"Probe module that sends a TCP SYN packet to a specific port. Possible "
+	"classifications are: synack and rst. A SYN-ACK packet is considered a "
+	"success and a reset packet is considered a failed response. "
+	"By default, TCP header options are set identically to the values for "
+	"Ubuntu (MSS, SACK permitted, Timestamp,  and WindowScale = 7). Use "
```
Comment (phillip-stephens) [inline src/probe_modules/module_tcp_synscan.c:270]:
```suggestion
	"linux (Ubuntu 23.04) (MSS, SACK permitted, Timestamp,  and WindowScale = 7). Use "
```

File: src/probe_modules/packet.c [hunk 4]
```diff
@@ -135,6 +142,130 @@ size_t set_mss_option(struct tcphdr *tcp_header)
...
+	last_opt[1] = 0x02; // set the length
+	last_opt += 2; // increment pointer
+	// EOL = 1 byte
+	last_opt[0] = 0x00; // kind for EOL
+	last_opt[1] = 0x00; // kind for EOL
+	last_opt += 2; // increment pointer
+	tcp_header->th_off += 1;
+	return tcp_header->th_off * 4;
+}
+
```
Comment (phillip-stephens) [inline src/probe_modules/packet.c:239]: Remove extra lines

File: src/probe_modules/module_tcp_synscan.c [hunk 5]
```diff
@@ -16,22 +16,60 @@
...
 #include "../fieldset.h"
+#include "module_tcp_synscan.h"
 #include "probe_modules.h"
 #include "packet.h"
 #include "validate.h"
 
-#define ZMAP_TCP_SYNSCAN_TCP_HEADER_LEN 24
-#define ZMAP_TCP_SYNSCAN_PACKET_LEN 58
+// defaults
+static uint8_t ZMAP_TCP_SYNSCAN_TCP_HEADER_LEN = 20;
```
Comment (zakird) [inline src/probe_modules/module_tcp_synscan.c:27]: Given that these aren't constants, I don't think they should be capitalized

Comment (phillip-stephens) [reply src/probe_modules/module_tcp_synscan.c:27]: Good call!

**Review by zakird — APPROVED** [review]

**Pull request merged**